packages/flutter/test/widgets/image_test.dart

```diff
@@ -35,6 +35,57 @@ void main() {
     imageCache.maximumSize = originalCacheSize;
   });
 
+  testWidgets('Verify Image does not use disposed handles', (WidgetTester tester) async {
+    final ValueNotifier<int> outerListenable = ValueNotifier<int>(0);
+    final ValueNotifier<int> innerListenable = ValueNotifier<int>(0);
+
+    bool imageLoaded = false;
+    MemoryImage image = MemoryImage(Uint8List.fromList(kTransparentImage));
+
+    Future<void> runAsyncAndIdle() async {
+      for (int i = 0; i < 20; ++i) {
+        await tester.runAsync(() => Future<void>.delayed(Duration.zero));
+        await tester.idle();
+      }
+    }
+
+    await tester.pumpWidget(ValueListenableBuilder<int>(
+      valueListenable: outerListenable,
+      builder: (BuildContext _, Object? __, Widget? ___) => Image(
+        image: image,
+        frameBuilder: (BuildContext _, Widget child, int? __, bool ___) {
+          if (((child as Semantics).child! as RawImage).image != null) {
```

Comment:
This is depending on implementation details that are not actually part of this test.

Instead, you could check if the `int? frameNumber` parameter `== 1`, which indicates the first (and in this case, only) frame of the image is loaded.

Reply:
This should also probably be setting a completer or a stream controller or something, so that it can be waited on without using the while loops below.